src/build-remote/build-remote.cc

```diff
@@ -0,0 +1,280 @@
+        decltype(supportedFeatures) supportedFeatures,
+        decltype(mandatoryFeatures) mandatoryFeatures) :
+        supportedFeatures{std::move(supportedFeatures)},
+        mandatoryFeatures{std::move(mandatoryFeatures)},
+        hostName{std::move(hostName)},
+        systemTypes{std::move(systemTypes)},
+        sshKey{std::move(sshKey)},
+        maxJobs{std::move(maxJobs)},
+        speedFactor{speedFactor == 0 ? 1 : std::move(speedFactor)},
+        enabled{true} {};
+};;
+
+static std::vector<machine> read_conf() {
+    auto conf = getEnv("NIX_REMOTE_SYSTEMS", SYSCONFDIR "/nix/machines");
+
+    auto machines = std::vector<machine>{};
+    auto confFile = std::ifstream{conf};
+    if (confFile.good()) {
+        confFile.exceptions(std::ifstream::badbit);
+        for (string line; getline(confFile, line);) {
+            chomp(line);
+            line.erase(std::find(line.begin(), line.end(), '#'), line.end());
+            if (line.empty()) {
+                continue;
+            }
+            auto tokens = tokenizeString<std::vector<string>>(line);
+            auto sz = tokens.size();
+            if (sz < 4) {
+                throw new FormatError(format("Bad machines.conf file %1%")
+                    % conf);
+            }
+            machines.emplace_back(tokens[0],
+                tokenizeString<std::vector<string>>(tokens[1], ","),
+                tokens[2],
+                stoull(tokens[3]),
+                sz >= 5 ? stoull(tokens[4]) : 1LL,
+                sz >= 6 ?
+                    tokenizeString<std::vector<string>>(tokens[5], ",") :
+                    std::vector<string>{},
+                sz >= 7 ?
+                    tokenizeString<std::vector<string>>(tokens[6], ",") :
+                    std::vector<string>{});
+        }
+    }
+    confFile.close();
+    return machines;
+}
+
+static string currentLoad;
+
+static int openSlotLock(const machine & m, unsigned long long slot) {
+    auto fn_stream = std::stringstream(currentLoad, std::ios_base::ate | std::ios_base::out);
+    fn_stream << "/";
+    for (auto t : m.systemTypes) {
+        fn_stream << t << "-";
+    }
+    fn_stream << m.hostName << "-" << slot;
+    return openLockFile(fn_stream.str(), true);
+}
+
+static char display_env[] = "DISPLAY=";
+static char ssh_env[] = "SSH_ASKPASS=";
+
+int main (int argc, char * * argv)
+{
+    return handleExceptions(argv[0], [&]() {
+        initNix();
+        /* Ensure we don't get any SSH passphrase or host key popups. */
+        if (putenv(display_env) == -1 ||
+            putenv(ssh_env) == -1) {
+            throw SysError("Setting SSH env vars");
+        }
+
+        if (argc != 4) {
+            throw UsageError("called without required arguments");
+        }
+
+        auto store = openStore();
+
+        auto localSystem = argv[1];
+        settings.maxSilentTime = strtoull(argv[2], NULL, 10);
+        settings.buildTimeout = strtoull(argv[3], NULL, 10);
```

Comment:
This prints `0`.
```c
unsigned long long res = strtoull("blah", NULL, 10);
printf("%llu\n", res);
```

The function returns `ULLONG_MAX` in case of overflow.